numpyro/infer/einstein/utils.py

```diff
@@ -0,0 +1,37 @@
+import jax
+import jax.numpy as jnp
+
+
+def posdef(m):
+    """  Map a matrix to a positive definite matrix, where all eigenvalues are >= 1e-5. """
+    mlambda, mvec = jnp.linalg.eigh(m)
+    if jnp.ndim(mlambda) >= 2:
+        mlambda = jax.vmap(lambda ml: jnp.diag(jnp.maximum(ml, 1e-5)), in_axes=tuple(range(jnp.ndim(mlambda) - 1)))(
+            mlambda)
+    else:
+        mlambda = jnp.diag(jnp.maximum(mlambda, 1e-5))
+    return mvec @ mlambda @ jnp.swapaxes(mvec, -2, -1)
+
+
+def sqrth(m):
+    """ Map a matrix to a positive definite matrix and square root it. """
```

Comment:
This square root is not `S @ S.T == m` right?

Reply:
It’s the hermitic square root, works for symmetric matrices ‘m’. I’ll update the doc string.

Reply:
Thanks, I didn't know that we can make square root symmetric this way. It's pretty nice math. :+1: